substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/service/ServiceFeature.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, 2024, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.core.genscavenge.service;
+
+import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
+import com.oracle.svm.core.feature.InternalFeature;
+import com.oracle.svm.core.jdk.RuntimeSupport;
+import com.oracle.svm.core.VMInspectionOptions;
+import org.graalvm.nativeimage.ImageSingletons;
+
+@AutomaticallyRegisteredFeature
+public class ServiceFeature implements InternalFeature {
```

Comment:
`ServiceThreadFeature`, `ServiceThreadStartupHook`, `ServiceThreadShutdownHook`, `ServiceThreadSupport`

Reply:
Based on https://github.com/oracle/graal/pull/9799#discussion_r1795872505, is it better to rename to `NotificationThreadFeature` , `NotificationThreadStartupHook`, etc ? I guess it depends on whether we plan to combine service and notifications together which might make sense if there aren't many services/notifications.

Reply:
Sounds good, let's use `Notification*` for now.

Reply:
I've  changed all prefixes to "Notification" and removed all uses of "Service" to be consistent.